### PR TITLE
feat: add set_learning_rate RPC to DTensorPolicyWorkerV2

### PR DESCRIPTION
**Pull request by GavinZhu-GMI**
# What does this PR do ?

Enables external orchestrators (service built on nemo RL) to dynamically set the optimizer learning rate on DTensor policy workers via run_all_workers_single_data("set_learning_rate", learning_rate=1e-5).

# Issues
 Without this, the LR is fixed at Policy init time and cannot be overridden per training step. Training frameworks that pass LR via optimizer step params (like Tinker's adam_params.learning_rate) had no way to propagate it to NeMo RL workers.

# Usage
* **You can potentially add a usage example below**

```python
       futures = policy.worker_group.run_all_workers_single_data(
           "set_learning_rate",
            learning_rate=learning_rate,
         )

       ray.get(futures)
       logger.info("Set learning rate to %s", learning_rate)
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ x ] Make sure you read and followed [Contributor guidelines](/NVIDIA-NeMo/RL/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you run the unit tests and functional tests locally? Visit our [Testing Guide](/NVIDIA-NeMo/RL/blob/main/docs/testing.md) for how to run tests
- [ ] Did you add or update any necessary documentation? Visit our [Document Development Guide](/NVIDIA-NeMo/RL/blob/main/docs/documentation.md) for how to write, build and test the docs.

# Additional Information

## Summary by CodeRabbit

* **New Features**
  * Added ability to dynamically update the optimizer's learning rate during training.

### CODE REVIEW

File: nemo_rl/models/policy/workers/dtensor_policy_worker_v2.py [hunk 1]
```diff
@@ -1054,6 +1054,12 @@ def offload_after_refit(self) -> None:
             f"GPU Memory after optimizer offload: {allocated:.2f}GB allocated, {reserved:.2f}GB reserved"
         )
 
+    def set_learning_rate(self, learning_rate: float) -> None:
+        """Set the learning rate on the optimizer's param groups."""
+        if self.optimizer is not None:
+            for pg in self.optimizer.param_groups:
+                pg["lr"] = learning_rate
+
```
[the remark this answers is not in the document]
Comment (GavinZhu-GMI) [reply nemo_rl/models/policy/workers/dtensor_policy_worker_v2.py:1062]: The validation is reasonable but overly cautious for an internal RPC. The optimizer is always initialized by the time training runs, and the LR comes from the SDK's validated AdamParams. Adding import math and raising exceptions for cases that can't happen in practice adds noise.

**Review by terrykong — COMMENTED** [review]
Hi @GavinZhu-GMI . Thanks for the contribution. Do you have any results or experiments that can help motivate why this API should be added/maintained moving forward? Any reason to not just shape the LR with the schedule?

There are many APIs we could add to the policy workers, so just trying to understand why this one.

**Comment by GavinZhu-GMI** [comment]
> Hi @GavinZhu-GMI . Thanks for the contribution. Do you have any results or experiments that can help motivate why this API should be added/maintained moving forward? Any reason to not just shape the LR with the schedule?
> 
> There are many APIs we could add to the policy workers, so just trying to understand why this one.

Hi @terrykong sorry for the late reply, here is a sum up.
 Tinker is an external orchestrator that uses NeMo RL in push-mode — the client controls the training loop ( just how it works). This creates a fundamental mismatch with NeMo RL's built-in OptimizerParamScheduler:
  1. Unknown total steps upfront — NeMo RL's scheduler requires lr_decay_iters and lr_warmup_iters at init time. Tinker clients don't know the total step count in advance; they **dynamically** decide when to train and at what LR each step.
  2. Client-side scheduling — Tinker cookbook recipes compute LR per-step on the client side (supporting linear, cosine, constant schedules) and pass the computed value via AdamParams(learning_rate=...) to the backend. The LR schedule lives in the orchestrator, not in NeMo RL.
